File: frontend/src/app/models/RotatingProxy.ts
export interface RotatingProxy {
  id: number;
  name: string;
  protocol: string;
  alive_proxy_count: number;
  listen_port: number;
  auth_required: boolean;
  auth_username?: string | null;
  last_rotation_at?: string | null;
  last_served_proxy?: string | null;
  created_at: string;
}

export interface CreateRotatingProxy {
  name: string;
  protocol: string;
  listen_port: number;
  auth_required: boolean;
  auth_username?: string | null;
  auth_password?: string | null;
}

export interface RotatingProxyNext {
  proxy_id: number;
  ip: string;
  port: number;
  username?: string | null;
  password?: string | null;
  has_auth: boolean;
  protocol: string;
}

File: backend/internal/api/dto/rotating_proxy.go
package dto

import "time"

type RotatingProxy struct {
	ID              uint64     `json:"id"`
	Name            string     `json:"name"`
	Protocol        string     `json:"protocol"`
	AliveProxyCount int        `json:"alive_proxy_count"`
	ListenPort      uint16     `json:"listen_port"`
	AuthRequired    bool       `json:"auth_required"`
	AuthUsername    string     `json:"auth_username,omitempty"`
	AuthPassword    string     `json:"auth_password,omitempty"`
	ListenHost      string     `json:"listen_host,omitempty"`
	ListenAddress   string     `json:"listen_address,omitempty"`
	LastRotationAt  *time.Time `json:"last_rotation_at,omitempty"`
	LastServedProxy string     `json:"last_served_proxy,omitempty"`
	CreatedAt       time.Time  `json:"created_at"`
}

type RotatingProxyCreateRequest struct {
	Name         string `json:"name"`
	Protocol     string `json:"protocol"`
	AuthRequired bool   `json:"auth_required"`
	AuthUsername string `json:"auth_username,omitempty"`
	AuthPassword string `json:"auth_password,omitempty"`
}

type RotatingProxyNext struct {
	ProxyID  uint64 `json:"proxy_id"`
	IP       string `json:"ip"`
	Port     uint16 `json:"port"`
	Username string `json:"username,omitempty"`
	Password string `json:"password,omitempty"`
	HasAuth  bool   `json:"has_auth"`
	Protocol string `json:"protocol"`
}
